src/main/java/cms/gov/madie/measure/repositories/MeasureSearchServiceImpl.java

```diff
@@ -210,36 +209,62 @@ public Page<MeasureListDTO> searchMeasuresByCriteria(
                 project(MeasureListDTO.class))
             .as("queryResults");
 
-    Aggregation pipeline = null;
+    Aggregation pipeline;
     if (appConfigService.isFlagEnabled(MadieFeatureFlag.MEASURE_SEARCH)) {
-      SortOperation sortOperation =
+      // Get all measureSetIds upfront based on the requested criteria
+      List<String> matchedMeasureSetIds =
+          mongoTemplate
+              .aggregate(
+                  newAggregation(
+                      lookupOperation, unwindOperation, matchOperation, group("measureSetId")),
+                  Measure.class,
+                  MeasureSetIdDTO.class)
+              .getMappedResults()
+              .stream()
+              .map(MeasureSetIdDTO::getId)
+              .collect(Collectors.toList());
+
+      MatchOperation matchMeasureSetIds =
+          match(Criteria.where("measureSetId").in(matchedMeasureSetIds));
+
+      SortOperation sortByVersionAndDraft =
           sort(Sort.by(Sort.Direction.DESC, "measureMetaData.draft", "version"));
-
-      GroupOperation groupOperation =
-          group("measureSetId").count().as("count").first("$$ROOT").as("selectedDoc");
-
-      AddFieldsOperation addFieldsOperation =
+      GroupOperation groupByMeasureSet = group("measureSetId").first("$$ROOT").as("selectedDoc");
+
+      // Lookup associated measures from the same set
+      LookupOperation lookupAssociatedMeasures =
+          LookupOperation.newLookup()
+              .from("measure")
+              .localField("selectedDoc.measureSetId")
+              .foreignField("measureSetId")
+              .as("associatedMeasures");
```

Comment:
A lookup can be thought of as a join, doing a join of the same table you are already querying seems like it could cause issues.